tokenizers/src/tokenizer/mod.rs

```diff
@@ -1216,7 +1206,7 @@ where
 
                 if add_special_tokens && n_added_tokens > 0 {
                     let params = TruncationParams {
-                        max_length: trunc.max_length - n_added_tokens,
+                        max_length: if n_added_tokens > trunc.max_length {0} else {trunc.max_length - n_added_tokens},
```

Comment:
```suggestion
                        max_length: trunc.max_length.checked_sub(n_added_tokens).unwrap_or(0)
```

NIT: I feel like this is more readable